File: PHP/Single Calls/pdf-with-added-image.php
<?php

$pdf_with_added_image_endpoint_url = 'https://api.pdfrest.com/pdf-with-added-image';

$data = array(
    'file' => new CURLFile('/path/to/file', 'file_name'),
    'image_file' => new CURLFile('/path/to/file','image/jpeg', 'file_name'),
    'output' => 'example_out',
    'x' => '10',
    'y' => '10',
    'page' => '1'
);

$headers = array(
    'Accept: application/json',
    'Content-Type: multipart/form-data',
    'Api-Key: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' // place your api key here
);

$ch = curl_init();

curl_setopt($ch, CURLOPT_URL, $pdf_with_added_image_endpoint_url);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
curl_setopt($ch, CURLOPT_POST, true);
curl_setopt($ch, CURLOPT_POSTFIELDS, $data);

print "Sending POST request to pdf-with-added-image endpoint...\n";
$response = curl_exec($ch);

print "Response status code: " . curl_getinfo($ch, CURLINFO_HTTP_CODE) . "\n";

if($response === false){
    print 'Error: ' . curl_error($ch) . "\n";
}else{
    print json_encode(json_decode($response), JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES);
    print "\n";
}

curl_close($ch);
?>

File: Python/Single Calls/pdf-with-added-image.py
from requests_toolbelt import MultipartEncoder
import requests
import json

pdf_with_added_image_endpoint_url = 'https://api.pdfrest.com/pdf-with-added-image'

mp_encoder_pdfWithAddedImage = MultipartEncoder(
    fields={
        'file': ('file_name', open('/path/to/file', 'rb'), 'application/pdf'),
        'image_file': ('file_name', open('/path/to/file', 'rb'), 'image/jpeg'),
        'output' : 'example_out',
        'x' : '10',
        'y' : '10',
        'page' : '1',
    }
)

headers = {
    'Accept': 'application/json',
    'Content-Type': mp_encoder_pdfWithAddedImage.content_type,
    'Api-Key': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' # place your api key here
}

print("Sending POST request to pdf-with-added-image endpoint...")
response = requests.post(pdf_with_added_image_endpoint_url, data=mp_encoder_pdfWithAddedImage, headers=headers)

print("Response status code: " + str(response.status_code))

if response.ok:
    response_json = response.json()
    print(json.dumps(response_json, indent = 2))
else:
    print(response.text)
